oci/table_oci_core_vnic_attachment.go

```diff
@@ -0,0 +1,423 @@
+			{
+				Name:        "vlan_tag",
+				Description: "The OCID of the VNIC.",
+				Type:        proto.ColumnType_INT,
+			},
+
+			// json column
+			{
+				Name:        "nsg_ids",
+				Description: "A list of the OCIDs of the network security groups that the VNIC belongs to.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getVnic,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getVnic,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getVnic,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getVnic,
+				Transform:   transform.From(vnicTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Id").Transform(ociRegionName),
+			},
+			{
+				Name:        "compartment_id",
+				Description: ColumnDescriptionCompartment,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CompartmentId"),
+			},
+			{
+				Name:        "tenant_id",
+				Description: ColumnDescriptionTenant,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     plugin.HydrateFunc(getTenantId).WithCache(),
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listVnicAttachments(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Trace("listVnicAttachments", "Compartment", compartment, "OCI_REGION", region)
+
+	equalQuals := d.KeyColumnQuals
+
+	// Return nil, if given compartment_id doesn't match
+	if equalQuals["compartment_id"] != nil && compartment != equalQuals["compartment_id"].GetStringValue() {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := coreComputeService(ctx, d, region)
+	if err != nil {
+		logger.Error("listVnicAttachments", "compute_service_error", err)
+		return nil, err
+	}
+
+	// Build request parameters
+	// Max limit isn't mentioned in the documentation
+	// Default limit is set as 1000
+	request := core.ListVnicAttachmentsRequest{
+		CompartmentId: types.String(compartment),
+		Limit:         types.Int(1000),
+		RequestMetadata: common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(),
+		},
+	}
+
+	// Check for additional filters
+	if equalQuals["availability_domain"] != nil {
```

Comment:
```suggestion
	if equalQuals["availability_domain"] != nil && equalQuals["availability_domain"].GetStringValue() != ""{
```
For the query `select * from oci_core_vnic_attachment where available_domain in ('abc', 'def')` the existing code will pass an empty string in param which may cause an error.

Reply:
Added the extra validation